### PR TITLE
[FIX] Verified property of user is always set to false if not supplied

### LINKED ISSUE #9718: Verified property of user is always set to false if not supplied
`verified` property in `user.emails`, is always set to false, if not supplied. Related in [this](https://github.com/RocketChat/Rocket.Chat.iOS/issues/962#issuecomment-349052238) issue.

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
@RocketChat/core 
Closes #9718
Added verification before update the user, because was being setting `verified` property always to false, if was not supplied.

### CODE REVIEW

**Comment by MarcosSpessatto** [comment]
Added a new endpoint to prevent any breaking change, which might happen if we modified the `users.update` enpoint directly.
Could you take a look please, @rodrigok ?

File: packages/rocketchat-lib/server/functions/saveUser.js [hunk 1]
```diff
@@ -167,7 +167,9 @@ RocketChat.saveUser = function(userId, userData) {
 			updateUser.$set.requirePasswordChange = userData.requirePasswordChange;
 		}
 
-		updateUser.$set['emails.0.verified'] = !!userData.verified;
+		if (userData.verified) {
+			updateUser.$set['emails.0.verified'] = true;
```
Comment (rodrigok) [inline packages/rocketchat-lib/server/functions/saveUser.js]: Can't I set as not verified? Shouldn't be more correct to check if `userData.verified` is boolean and then allow to change, that way we do not require to pass that information but allow admins to revert some mistake.

File: packages/rocketchat-api/server/v1/users.js [hunk 2]
```diff
@@ -260,13 +260,48 @@ RocketChat.API.v1.addRoute('users.update', { authRequired: true }, {
...
+			realname: this.bodyParams.data.name,
+			username: this.bodyParams.data.username,
+			newPassword: this.bodyParams.data.newPassword
+		};
+
+		if (this.bodyParams.data.actualPassword) {
+			userData.typedPassword = SHA256(this.bodyParams.data.actualPassword);
+		}
+
+		Meteor.runAsUser(this.bodyParams.userId, () => Meteor.call('saveUserProfile', userData, this.bodyParams.customFields));
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/users.js]: That way you are allowing anyone to change other's user information, should run as the authenticated user

File: packages/rocketchat-api/server/v1/users.js [hunk 3]
```diff
@@ -260,13 +260,48 @@ RocketChat.API.v1.addRoute('users.update', { authRequired: true }, {
...
 
+RocketChat.API.v1.addRoute('users.updateOwnBasicInfo', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, {
+			userId: String,
+			data: Match.ObjectIncluding({
+				email: Match.Maybe(String),
+				name: Match.Maybe(String),
+				username: Match.Maybe(String),
+				actualPassword: Match.Maybe(String),
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/users.js]: Wouldn't be better `currentPassword` rather than `actualPassword `?

**Pull request merged**